source/Irrlicht/CIrrDeviceSDL.cpp

```diff
@@ -712,25 +703,30 @@ bool CIrrDeviceSDL::run()
 			Close = true;
 			break;
 
-		case SDL_ACTIVEEVENT:
-			if (SDL_event.active.state == SDL_APPACTIVE)
-				WindowMinimized = (SDL_event.active.gain!=1);
-			break;
-
-		case SDL_VIDEORESIZE:
-			if ((SDL_event.resize.w != (int)Width) || (SDL_event.resize.h != (int)Height))
+		case SDL_WINDOWEVENT:
+			switch (SDL_event.window.event)
 			{
-				Width = SDL_event.resize.w;
-				Height = SDL_event.resize.h;
+			case SDL_WINDOWEVENT_MAXIMIZED:
+				WindowMinimized = true;
+				break;
+			case SDL_WINDOWEVENT_RESTORED:
+				WindowMinimized = false;
+				break;
+			case SDL_WINDOWEVENT_RESIZED:
+				if ((SDL_event.window.data1 != (int)Width) || (SDL_event.window.data2 != (int)Height))
+				{
+					Width = SDL_event.window.data1;
+					Height = SDL_event.window.data2;
 #ifdef _IRR_EMSCRIPTEN_PLATFORM_
-				Screen = SDL_SetVideoMode( 0, 0, 32, SDL_Flags ); // 0,0 will use the canvas size
+					//SDL_CreateWindowAndRenderer(0, 0, SDL_Flags, &Window, &Renderer); // 0,0 will use the canvas size
 #else //_IRR_EMSCRIPTEN_PLATFORM_
- 				Screen = SDL_SetVideoMode( Width, Height, 0, SDL_Flags );
+ 					//Window = SDL_CreateWindow("", SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED, Width, Height, SDL_Flags);
 #endif //_IRR_EMSCRIPTEN_PLATFOR

```

Comment:
lines should be removed if no longer needed
(also applies in a few more places further down)